File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void kSmallest(TreeNode* root, priority_queue<int>& q){
        q.push(-(root->val));
        if(root->left)
        kSmallest(root->left,q);
        if(root->right)
        kSmallest(root->right,q);

    }
    int kthSmallest(TreeNode* root, int k) {
        priority_queue<int>q;
        kSmallest(root,q);
        for(int i =0;i<k-1;i++){
            q.pop();
        }
        return -q.top();
    }
};

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int res = 0;
    int g ;
    void inorder(TreeNode root, int k ){
        if(root == null || g== 0){
            return;
        }
        inorder(root.left,k);
        g--;
        if(g== 0){
            res=root.val;
            return;
        }
        inorder(root.right,k);
    }
    public int kthSmallest(TreeNode root, int k) {
        g=k;
        inorder(root, k);
        return res;
    }
}